samples/sample_hevc_fei/src/sample_hevc_fei.cpp

```diff
@@ -596,13 +650,26 @@ mfxStatus CheckOptions(const sInputParams& params, const msdk_char* appName)
         PrintHelp(appName, "Unsupported NumMvPredictorsL1 value (must be in range [0,4])");
         return MFX_ERR_UNSUPPORTED;
     }
-    if (!(params.encodeCtrl.NumFramePartitions == 1 ||
-          params.encodeCtrl.NumFramePartitions == 2 ||
-          params.encodeCtrl.NumFramePartitions == 4 ||
-          params.encodeCtrl.NumFramePartitions == 8 ||
-          params.encodeCtrl.NumFramePartitions == 16  ))
+    std::array<mfxU16, 4> nfp_values = {1, 2, 4, 8};
+    auto it = std::find_if(nfp_values.begin(), nfp_values.end(), [&params](const mfxU16 nfp_val)
```

Comment:
this one is not find_if, but find, you don't need such complicated machinery to find value in array.
Anyway, see comment above